ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -208,24 +208,34 @@ private Callback<Void> securityProcessRequestCallback() {
      */
     private Callback<Void> securityPostProcessRequestCallback(BlobInfo blobInfo) {
       return buildCallback(frontendMetrics.putSecurityPostProcessRequestMetrics, securityCheckResult -> {
-        if (RestUtils.isNamedBlobStitchRequest(restRequest)) {
-          if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
-            accountAndContainerInjector.injectDatasetForNamedBlob(restRequest);
-            frontendMetrics.addDatasetVersionRate.mark();
-            addDatasetVersion(blobInfo.getBlobProperties(), restRequest);
-          }
-          RetainingAsyncWritableChannel channel =
-              new RetainingAsyncWritableChannel(frontendConfig.maxJsonRequestSizeBytes);
-          restRequest.readInto(channel, fetchStitchRequestBodyCallback(channel, blobInfo));
+        if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
+          accountAndContainerInjector.injectDatasetForNamedBlob(restRequest);
+          frontendMetrics.addDatasetVersionRate.mark();
+          addDatasetVersion(blobInfo.getBlobProperties(), restRequest);
+        }
+        idConverter.detectConflict(restRequest, detectConflictCallback(blobInfo));
```

Comment:
Hmm, this will result in checking for conflicts for regular named blob uploads as well, i.e. not just `stitch` requests? Wondering if it may increase in latency of regular uploads due to additional call to mysql db. Would it be better to check for conflicts only for `stitch` requests?